### PR TITLE
[MM-39745] Disable channel header buttons temporarily when the RHS closes

### PR DESCRIPTION
**Pull request by mickmister**
#### Summary

With the current channel header icon placement in relation to the RHS, users have been accidentally starting zoom meetings because the zoom icon is placed at the same spot as the close-x of the RHS thread view. If the user clicks to close the RHS twice, the zoom button gets clicked.

The channel header icons will be moved to the App Bar in the near future, which will make it so this is no longer a problem, but this PR is a solution to avoid this issue in the meantime. This PR makes it so when user closes the RHS, the channel header icons are disabled for a short period of time, to avoid the double-click issue.

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-39745

### CODE REVIEW

File: plugins/channel_header_plug/channel_header_plug.tsx
Comment (mickmister) [inline plugins/channel_header_plug/channel_header_plug.tsx:142]: I tried to implement this with `getDerivedStateFromProps`, but since we need to perform a side effect, it couldn't be done with a static method.

File: plugins/channel_header_plug/channel_header_plug.test.tsx
Comment (larkox) [inline plugins/channel_header_plug/channel_header_plug.test.tsx:86]: Does it make sense add any test with `sidebarOpen={true}`?

File: plugins/channel_header_plug/channel_header_plug.tsx
Comment (larkox) [inline plugins/channel_header_plug/channel_header_plug.tsx:163]: Is this necessary? This is for when the buttons are collapsed and you select one element, right?

File: plugins/channel_header_plug/channel_header_plug.tsx
Comment (larkox) [inline plugins/channel_header_plug/channel_header_plug.tsx:185]: Apart of not allowing to hit the buttons, do we want any visual feedback of they become unavailable? Or not? (I guess not, but wanted to ask the question just in case)

File: plugins/channel_header_plug/channel_header_plug.tsx
Comment (larkox) [inline plugins/channel_header_plug/channel_header_plug.tsx:139]: Probably nothing to worry about, but opening and closing really fast may cause the buttons being enabled too soon. Nevertheless, since we are talking about fractions of seconds here, probably it is something we don't have to worry about.

File: plugins/channel_header_plug/channel_header_plug.tsx
Comment (larkox) [inline plugins/channel_header_plug/channel_header_plug.tsx:141]: Also something we shouldn't worry about probably, but if the channel_header_plug component gets unmounted before the timer kicks in, we may have a "trying to set state on unmounted object" error.

I don't foresee how the channel_header_plug component may be unmounted, specially in such short notice, but wanted to raise the concern just in case.

**Review by larkox — APPROVED** [review]
LGTM, just some nitpick here and there.

File: plugins/channel_header_plug/channel_header_plug.tsx
Comment (hmhealey) [inline plugins/channel_header_plug/channel_header_plug.tsx:137]: To avoid the extra renders caused by storing this in state and fix that warning, I think we could store `disableButtonsClosingRHS` as a member variable. It doesn't affect rendering anyway, so it doesn't need to be in state.

**Review by hmhealey — CHANGES_REQUESTED** [review]

File: plugins/channel_header_plug/channel_header_plug.test.tsx
Comment (mickmister) [reply plugins/channel_header_plug/channel_header_plug.test.tsx]: N/A since the variable was moved out of the component's state

File: plugins/channel_header_plug/channel_header_plug.tsx
Comment (mickmister) [reply plugins/channel_header_plug/channel_header_plug.tsx]: Good call, removed this block

File: plugins/channel_header_plug/channel_header_plug.tsx
Comment (mickmister) [reply plugins/channel_header_plug/channel_header_plug.tsx]: N/A since this variable has been moved out of the component's state

**Review by hmhealey — APPROVED** [review]

**Comment by mickmister** [comment]
@DHaussermann This is ready for review

**Comment by mickmister** [comment]
@DHaussermann Note that this PR works by
- Waiting for the RHS to completely finish closing
- Then disabling the channel header component actions for 500ms, meaning clicking them during this time will result in no action. i.e. no zoom meeting will be started
- After the 500ms is done, you can click the channel header components and start a zoom meeting

**Comment by DHaussermann** [comment]
Tested and passed 
- This change was successfully implemented. Functional testing confirmed a 0.5 second delay.
- The  0.5s delay begins once the RHS is fully closed
- Deployed locally and tested with various plugins to ensure the click is ignored
- Tested on browser and desktop

As a precaution - I'd like to see the Cypress tests run on this PR before we merge to ensure there is no maintenance being introduced.    
I will approve once I get this done.

**Comment by mickmister** [comment]
> As a precaution - I'd like to see the Cypress tests run on this PR before we merge to ensure there is no maintenance being introduced. I will approve once I get this done.

@DHaussermann Is this something you need help with are can you do this locally?

**Review by DHaussermann — APPROVED** [review]
The tooling for the Cypress tests is working fine. No need for anything locally, this PR was just backed up in a queue.

Reviewed Cy test results. Pass rate is comparable to before this PR. Reviewed all failing tests and none show indications of being related to RHS or channel header. I see no reason to suspect this will introduce any maintenance work. 

Tested and passed as per comment above. 

LGTM!

@mickmister please merge when you can.

**Pull request merged**
